src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -499,7 +520,17 @@ def makeTransferRucio(self, wflow, dataIn, dids, dataSize, grouping, copies, nod
             self.logger.info("Creating rule for workflow %s with %d DIDs in container %s, RSEs: %s, grouping: %s",
                              wflow.getName(), len(dids), dataIn['name'], rseExpr, grouping)
             try:
-                res = self.rucio.createReplicationRule(dids, rseExpr, **ruleAttrs)
+                # make decision about current workflow, if it is new request we'll create
+                # new replication rule, otherwise we'll move replication rule
+                if wflow.updatedWorkflow:
+                    rules = self.rucio.listDataRules(wflow.getName())
+                    for rdoc in rules:
+                        for rid in rdoc['ruleIds']:
+                            # the moveReplcationRule will raise different exceptions
+                            # if move operation is not normal
+                            self.rucio.moveReplicationRule(rid, rseExpr, **ruleAttrs)
```

Comment:
It looks like you should pass only the rule id and rse expression to this call.

Reply:
ok